src/Client/Phiremock.php

```diff
@@ -36,14 +36,22 @@ class Phiremock
      * @var int
      */
     private $port;
+    /**
+     * @var array
+     */
+    private $clientConfig = [
+        'http_errors' => false,
+    ];
 
     public function __construct(
         $host = 'localhost',
         $port = 8080,
         RemoteConnectionInterface $remoteConnection = null
     ) {
         if (!$remoteConnection) {
-            $remoteConnection = new GuzzleConnection();
+            $remoteConnection = new GuzzleConnection(
+                new \GuzzleHttp\Client($this->clientConfig)
```

Comment:
`self::checkErrorResponse()` was never called because exception was already thrown by Guzzle on 4xx and 5xx responses. Fixed.